components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -28,6 +28,11 @@ service WorkspacesService {
     //   FAILED_PRECONDITION: if there's no running instance
     rpc StopWorkspace(StopWorkspaceRequest) returns (stream StopWorkspaceResponse) {}
 
+    // SendHeartbeat sends a heartbeat signal to a running workspace.
+    rpc SendHeartbeat(SendHeartbeatRequest) returns (SendHeartbeatResponse) {}
+
+    // SendCloseHeartbeat sends a close heartbeat signal to a running workspace.
+    rpc SendCloseHeartbeat(SendCloseHeartbeatRequest) returns (SendCloseHeartbeatResponse) {}
```

Comment:
I would call it `SendClose` or `SendCloseSignal`. The point is that it is not heartbeating fiction anymore.